### PR TITLE
Created the class DecimalUtils and added the Unit Tests

### PR DESCRIPTION
**Pull request by MarcosCGlobant**
Created the class DecimalUtils so the format of the result can reach the Unit Tests.

Added the Unit Tests for the class CalculatorPresenter and Calculator Model, mocking the View.

The picture show that all the tests passed.

Test Passed

### CODE REVIEW

File: app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+        presenter.onOperatorPressed(PLUS);
+        assertNotEquals(EMPTY_STRING, model.getOperator());
+        verify(mockedView).showOperationPressed(PLUS);
+        verify(mockedView).handleDot(true);
+    }
+
+    @Test
+    public void onOperatorPressedWithSecondNumberShowOperatorAndSetOnModel() {
+        when(decimalFormatForResults.getResultWithNewFormat(100.0)).thenReturn("100");
+        model.setFirstNumber(NUMBER_TEST);
```
Comment (BQuinteros) [inline app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java:92]: pls create a String to put "100"

File: app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        model.setSecondNumber(EMPTY_STRING);
+        assertEquals(EMPTY_STRING, model.getSecondNumber());
+        assertNotEquals(EMPTY_STRING, model.getOperator());
+        verify(mockedView).showOperationPressed(MINUS);
+        verify(mockedView).handleDot(true);
+    }
+
+    @Test
+    public void onEqualsPressedWithFirstAndSecondNumberAndWithoutThoseNumbersBeenJustDots() {
+        when(decimalFormatForResults.getResultWithNewFormat(100.0)).thenReturn("100");
```
Comment (BQuinteros) [inline app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java]: pls create a String to put "100"

File: app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+        presenter.onNumberButtonPress(DOT_BUTTON);
+        assertEquals(DOT_BUTTON, model.getFirstNumber());
+        verify(mockedView).showNumberPressed(DOT_BUTTON);
+        verify(mockedView).handleDot(false);
+        verify(mockedView).handleOperations(true);
+    }
+
+    @Test
+    public void onEqualsPressedWithFirstAndSecondNumberWithoutThoseNumbersBeenJustDotsAndWithAMinusOperator() {
+        when(decimalFormatForResults.getResultWithNewFormat(0.0)).thenReturn("0");
```
Comment (BQuinteros) [inline app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java]: pls create a String to put "0"

File: app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+        presenter.onEqualsButtonPressed();
+        verify(mockedView).showResult(NUMBER_ZERO);
+        assertEquals(EMPTY_STRING, model.getFirstNumber());
+        assertEquals(EMPTY_STRING, model.getOperator());
+        assertEquals(EMPTY_STRING, model.getSecondNumber());
+    }
+
+    @Test
+    public void onEqualsPressedWithFirstAndSecondNumberWithoutThoseNumbersBeenJustDotsAndWithAMultiplyOperator() {
+        when(decimalFormatForResults.getResultWithNewFormat(25.0)).thenReturn("25");
```
Comment (BQuinteros) [inline app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java]: pls create a String to put "25"

File: app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+        presenter.onEqualsButtonPressed();
+        verify(mockedView).showResult(MULTIPLY_RESULT);
+        assertEquals(EMPTY_STRING, model.getFirstNumber());
+        assertEquals(EMPTY_STRING, model.getOperator());
+        assertEquals(EMPTY_STRING, model.getSecondNumber());
+    }
+
+    @Test
+    public void onEqualsPressedWithFirstAndSecondNumberWithoutThoseNumbersBeenJustDotsAndWithADivideOperator() {
+        when(decimalFormatForResults.getResultWithNewFormat(5.0)).thenReturn("5");
```
Comment (BQuinteros) [inline app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java]: pls create a String to put "5"

Comment (MarcosCGlobant) [reply app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java]: done, thanks

File: app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java (hunk 4 above)
Comment (MarcosCGlobant) [reply app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java]: done, thanks

File: app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java (hunk 3 above)
Comment (MarcosCGlobant) [reply app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java]: done, thanks

File: app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java (hunk 2 above)
Comment (MarcosCGlobant) [reply app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java]: done, thanks

File: app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java (hunk 1 above)
Comment (MarcosCGlobant) [reply app/app/src/test/java/com/globant/calculator/android/mvp/presenter/CalculatorPresenterTest.java:92]: done, thanks

File: app/app/src/test/java/com/globant/calculator/android/OperationUnitTest.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package com.globant.calculator.android;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
```
Comment (mauroLonghi) [inline app/app/src/test/java/com/globant/calculator/android/OperationUnitTest.java:5]: Pls remove the wildcare imports.

File: app/app/src/test/java/com/globant/calculator/android/OperationUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+    @Test
+    public void just_dot_as_second_value() {
+        assertEquals("Invalid operation", 2 + ".");
+    }
+
+    @Test
+    public void divided_by_zero() {
+        // assertArrayEquals("Invalid operation", 2 / 0);
```
Comment (mauroLonghi) [inline app/app/src/test/java/com/globant/calculator/android/OperationUnitTest.java:46]: If this method is commented please remove the method or include a descriptive comment.

**Pull request closed without merging**